sled-agent/src/rack_setup/service.rs

```diff
@@ -1117,126 +1139,84 @@ impl ServiceInner {
-                        &rack_network_config.uplink_port_speed,
-                    ),
-                },
-                addrs: vec![addr],
-            };
+                let (ipv6_entry, dpd_port_settings, port_id) =
+                    self.build_uplink_config(uplink_config)?;
 
-            dpd_port_settings.links.insert(link_id.to_string(), link_settings);
+                self.wait_for_dendrite(dpd).await;
 
-            let port_id: PortId = rack_network_config.uplink_port.parse()
-                .map_err(|e| SetupServiceError::BadConfig(
-                        format!("could not use value provided to rack_network_config.uplink_port as PortID: {e}")
-                ))?;
+                info!(self.log, "Configuring boundary services loopback address on switch"; "config" => format!("{ipv6_entry:#?}"));
+                dpd.loopback_ipv6_create(&ipv6_entry).await.map_err(|e| {
+                    SetupServiceError::Dendrite(format!(
+                        "unable to create inital switch loopback address: {e}"
+                    ))
+                })?;
 
-            let nexthop = Some(rack_network_config.gateway_ip);
+                info!(self.log, "Configuring default uplink on switch"; "config" => format!("{dpd_port_settings:#?}"));
+                dpd.port_settings_apply(&port_id, &dpd_port_settings)
+                    .await
+                    .map_err(|e| {
+                        SetupServiceError::Dendrite(format!(
+                    "unable to apply initial uplink port configuration: {e}"
+                ))
+                    })?;
 
-            dpd_port_settings.v4_routes.insert(
-                Ipv4Cidr { prefix: "0.0.0.0".parse().unwrap(), prefix_len: 0 }
-                    .to_string(),
-                RouteSettingsV4 {
-                    link_id: link_id.0,
-                    vid: rack_network_config.uplink_vid,
-                    nexthop,
-                },
-            );
+                info!(
+                    self.log,
+                    "advertising boundary services loopback address"
+                );
+                let zone_addr =
+                    switch_mgmt_addrs.get(&uplink_config.switch).unwrap();
 
-            loop {
-                info!(self.log, "Checking dendrite uptime");
-                match dpd.dpd_uptime().await {
-                    Ok(uptime) => {
-                        info!(self.log, "Dendrite online"; "uptime" => uptime.to_string());
-                        break;
-                    }
-                    Err(e) => {
-                        info!(self.log, "Unable to check Dendrite uptime"; "reason" => format!("{e}"));
-                    }
+                let ddmd_addr = SocketAddrV6::new(*zone_addr, DDMD_PORT, 0, 0);
+                let ddmd_client = DdmAdminClient::new(&self.log, ddmd_addr)?;
+                ddmd_client.advertise_prefix(Ipv6Subnet::new(
+                    BOUNDARY_SERVICES_ADDR.parse().unwrap(),
```

Comment:
Can we just grab this out of `ipv6_entry` to limit places we're hardcoding constants?